### DISCUSSION TITLE
Installing on project requiring newer version of System.Text.Json (>= 8)

### DISCUSSION THREAD

**Discussion by martimors** [xunit/xunit#2869] [Question] [answered]
Hi all,

I want to try out the xunit.v3 alpha builds, because they have some very welcome features such as the new `Explicit` option for tests, allowing deterring tests to only run when explicitly required.

However, my project depends on another library that bizarrely pins `System.Text.Json` to `≥8.0.0`. This project (`xunit`) pins that same dependency to exactly `=6.0.0` (see [here](https://github.com/xunit/xunit/blob/main/src/Versions.props#L32)).

Could someone help me understand 1) the reasoning for this restrictive pin on that package, and 2) if there is anything I can do to get `v3` running in my environment?

**Comment by bradwilson** [comment, marked as answer]
The issue was caused by my desire to use a strong version in the .csproj for `xunit.v3.common` (because I don't want `dotnet outdated` to suggest I upgrade it, since I stick it there as what was simultaneously released with .NET 6, which is our target framework, so it makes sense for us to stick with the lowest common denominator).

I've added a hand-crafted .nuspec file so that the stuck version doesn't translate to the package. The fix is available in `0.1.1-pre.354`.

**Reply by martimors** [reply]
Amazing stuff. Thanks for a great library. Very excited for v3 🤘
